packages/runtime/src/builder/serialization/control-serialization.ts

```diff
@@ -866,6 +870,52 @@ function deserializeRichTextControl(
   return { ...serializedControl, options: deserializedOptions }
 }
 
+type RichTextV2PluginControlDefinition<T extends ControlDefinition> = {
+  definition: T
+}
+
+type RichTextV2Plugin = {
+  controls?: RichTextV2PluginControlDefinition<ControlDefinition>[]
+}
+
+type RichTextV2Config = {
+  plugins?: RichTextV2Plugin[]
+  mode?: RichTextV2Mode
+}
```

Comment:
Should this type and the other 2 above this be in a separate file? They don't seem to be related to serialization specifically

Reply:
From my scan of the file, it seemed like there were a lot of partial types that were the result of serialization written inline above the serialization functions. I guess I'm confused about where else I would put these partial types.

Would it be better if I defined them here, but based them on the originals using the `Omit` keyword to remove anything that won't be passed to the builder?

Reply:
Ah, it threw me off because it didn’t have “serializable” on the name and I thought these were the control types. We should be explicit with naming to avoid confusion